sklearn/cross_decomposition/pls_.py

```diff
@@ -271,6 +271,10 @@ def fit(self, X, Y):
 
         # NIPALS algo: outer loop, over components
         for k in range(self.n_components):
+            if np.all(np.dot(Yk.T, Yk) < np.finfo(np.double).eps):
+                # Yk constant
+                warnings.warn('Y constant at iteration %s' % k)
+                break
```

Comment:
I fixed this because the common tests broke with the previous behavior. i don't like estimators breaking on trivial datasets. It tries to regress Y from X iteratively, then adjusts Y to be the residual. If the residual is constant 0, you can't learn any more. In the current implementation, you go on, but divide by the squared norm of Y, which results in nan errors.
I'm fine with your phrasing and test. The np.dot is used here in other places, and the difference is just that you are computing an additional squarerroot, right?